AxelarHandler/src/GoFastHandler.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import {IERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Ownable} from "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+
+import {ISwapRouter02} from "./interfaces/ISwapRouter02.sol";
+import {IFastTransferGateway} from "./interfaces/IFastTransferGateway.sol";
+
+contract GoFastHandler is Ownable {
+    using SafeERC20 for IERC20;
+
+    ISwapRouter02 public swapRouter;
+    IFastTransferGateway public fastTransferGateway;
+
+    constructor(address _swapRouter, address _fastTransferGateway) {
+        swapRouter = ISwapRouter02(_swapRouter);
+        fastTransferGateway = IFastTransferGateway(_fastTransferGateway);
+    }
+
+    function setSwapRouter(address _swapRouter) public onlyOwner {
+        swapRouter = ISwapRouter02(_swapRouter);
+    }
+
+    function setFastTransferGateway(address _fastTransferGateway) public onlyOwner {
+        fastTransferGateway = IFastTransferGateway(_fastTransferGateway);
+    }
+
+    function swapAndSubmitOrder(
+        address tokenIn,
+        uint256 swapAmountIn,
+        bytes memory swapCalldata,
+        uint256 executionFeeAmount,
+        uint256 solverFeeBPS,
+        bytes32 sender,
+        bytes32 recipient,
+        uint32 destinationDomain,
+        uint64 timeoutTimestamp,
+        bytes calldata destinationCalldata
+    ) public payable returns (bytes32) {
+        require(executionFeeAmount != 0, "execution fee cannot be zero");
```

Comment:
@thal0x if we remove this check i think we have a potential fund loss bug if the swapCallData ouput denom does not match the fast transfer gateway asset. can we rewrite things to avoid that possibility here? 

it also seems like we should just make the contract upgradeable - seems like cheap insurance against stuck funds in general. wdyt?